core/services/offchainreporting/OCRContract.go

```diff
@@ -0,0 +1,218 @@
+package offchainreporting
+
+import (
+	"context"
+	"math/big"
+	"strings"
+	"sync"
+	"time"
+
+	ethereum "github.com/ethereum/go-ethereum"
+	gethCommon "github.com/ethereum/go-ethereum/common"
+
+	"github.com/smartcontractkit/chainlink/core/services/eth"
+	"github.com/smartcontractkit/chainlink/core/store/models"
+	"github.com/smartcontractkit/chainlink/offchainreporting/confighelper"
+	"github.com/smartcontractkit/chainlink/offchainreporting/gethwrappers/offchainaggregator"
+	ocrtypes "github.com/smartcontractkit/chainlink/offchainreporting/types"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/pkg/errors"
+)
+
+var (
+	OCRContractConfigSet = getConfigSetHash()
+)
+
+type (
+	OCRContract struct {
+		ethClient        eth.Client
+		configSetChan    chan offchainaggregator.OffchainAggregatorConfigSet
```

Comment:
Is this used anywhere?

Reply:
Apparently not, it appears to be a holdover.

This file has been worked on by multiple people at multiples times it seems and has a bit of a patchwork feel to it.

Reply:
Agreed, glad we're cleaning it up now for core integration :)